From b61ffe92b91bf1bb7d07813410ea406c8871051b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mika=20Bostr=C3=B6m?= <bostik@odin.bostik.iki.fi>
Date: Fri, 16 Dec 2011 12:20:06 +0200
Subject: [PATCH 1/3] Fix build in dedicated buildroot

Do not assume that system paths are writable: remove DESTDIR rules which
point directly to Qt's installation path.

Add new file, src/threed/includes.pri which adds the individual
threed/MODULE and quick3d header paths to INCLUDEPATH. This allows to
include all the headers with a single easy addition.

Modify all module builds to include this new file so they find in-tree
headers and thus allow the build to work.

Fix quick3d build by telling it to use build-tree's header and library
location for Qt3D.

And finally, add 'quick' to included Qt modules. This is not strictly
necessary but having naked declarative module only will issue a
complaint about the issue and suggests doing such a modification.

Together these fixes allow to build qtquick3d in a cleaner fashion.
---
 src/imports/shapes/shapes.pro              |    5 +++--
 src/imports/threed/threed.pro              |    5 +++--
 src/plugins/imageformats/tga/tga.pro       |    2 +-
 src/plugins/sceneformats/assimp/assimp.pro |    2 +-
 src/plugins/sceneformats/bezier/bezier.pro |    2 +-
 src/quick3d/quick3d.pro                    |    3 +++
 src/threed/includes.pri                    |   15 +++++++++++++++
 7 files changed, 27 insertions(+), 7 deletions(-)
 create mode 100644 src/threed/includes.pri

diff --git a/src/threed/includes.pri b/src/threed/includes.pri
new file mode 100644
index 0000000..a7893e5
--- /dev/null
+++ b/src/threed/includes.pri
@@ -0,0 +1,15 @@
+# Add everything in threed/* to include paths
+INCLUDEPATH += $$PWD/../quick3d
+INCLUDEPATH += $$PWD/scene
+INCLUDEPATH += $$PWD/effects
+INCLUDEPATH += $$PWD/geometry
+INCLUDEPATH += $$PWD/arrays
+INCLUDEPATH += $$PWD/global
+INCLUDEPATH += $$PWD/math3d
+INCLUDEPATH += $$PWD/materials
+INCLUDEPATH += $$PWD/textures
+INCLUDEPATH += $$PWD/painting
+INCLUDEPATH += $$PWD/surfaces
+INCLUDEPATH += $$PWD/viewing
+INCLUDEPATH += $$PWD/graphicsview
+
diff --git a/src/imports/threed/includes.pri b/src/imports/threed/includes.pri
new file mode 100644
#index 0000000..a7893e5
--- /dev/null
+++ b/src/imports/threed/includes.pri
@@ -0,0 +1,15 @@
+# Add everything in threed/* to include paths
+INCLUDEPATH += $$PWD/../../quick3d
+INCLUDEPATH += $$PWD/scene
+INCLUDEPATH += $$PWD/effects
+INCLUDEPATH += $$PWD/geometry
+INCLUDEPATH += $$PWD/arrays
+INCLUDEPATH += $$PWD/global
+INCLUDEPATH += $$PWD/math3d
+INCLUDEPATH += $$PWD/materials
+INCLUDEPATH += $$PWD/textures
+INCLUDEPATH += $$PWD/painting
+INCLUDEPATH += $$PWD/surfaces
+INCLUDEPATH += $$PWD/viewing
+INCLUDEPATH += $$PWD/graphicsview
+
-- 
1.7.7.3

